nemo/collections/nlp/models/language_modeling/megatron_base_prompt_learning_model.py

```diff
@@ -133,14 +132,16 @@ def init_model(self, cfg: DictConfig, trainer: Trainer):
         self.pad_token_id = self.tokenizer.pad_id if self.tokenizer.pad_id is not None else self.tokenizer.unk_id
         self.decoder_seq_length = cfg.get('decoder_seq_length', 40)
 
-        if self.trainer.precision == 'bf16':
+        # Check for first few chars to be '16', '32' or 'bf16' as Precision can also be '16-mixed', '32-true' or 'bf16-mixed'
+        # along with 16, 32 or bf16 in case of PTL >= 2.0
+        if self.trainer.precision[0:4] == 'bf16':
```

Comment:
not sure if its only me, but this indexing feels slightly hacky, can we change it something like what you have in some code earlier: `cfg.trainer.precision in [16, '16-mixed', '16-true']`

Reply:
Fixed.